verifier/src/lib.rs

```diff
@@ -564,4 +623,55 @@ mod tests {
             "Forced failure via `AlwaysFalse`"
         );
     }
+
+    #[test]
+    fn display_of_notted_success() {
+        let failure = CtOption::new(
+            NotError::new(CtOption::new(
+                OrError::new(
+                    CtOption::new(VerificationError::General, 0.into()),
+                    CtOption::new(
+                        VerificationError::MiscellaneousSelectMismatch {
+                            expected: 2.into(),
+                            actual: 3.into(),
+                        },
+                        1.into(),
+                    ),
+                ),
+                1.into(),
+            )),
+            1.into(),
+        );
+        let displayable = failure.display();
+        let expected = r#"
+            NotError {
+                inner: OrError {
+                    left: Passed,
+                    right: The MiscellaneousSelect did not match expected:MiscellaneousSelect(2) actual:MiscellaneousSelect(3),
```

Comment:
Shown here and in the next test, the displaying of errors doesn't traverse down a "passed" branch.
It seems like the `Not` may be difficult to parse if we choose the current display path.

One thought is to update the `Display` implementation to do something like the below
```console
NotError {
    inner: Passed: OrError {
        left: Passed: (The attributes did not match expected:30  actual:30),
        right: Failed: (The MiscellaneousSelect did not match expected:MiscellaneousSelect(2) actual:MiscellaneousSelect(3)),
    }
}
```
the above example feels hard to parse, especially with the negation of `Not`, and the `Passed` followed by a failure message. 
it could be that the error messages should be updated to be more nuetral, i.e instead of "X did not match" it could say "expected a to match b"

Reply:
This is being discussed on #31